sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -2259,3 +2259,37 @@ def _weights(dist):
     est = KNeighborsRegressor(n_neighbors=1, algorithm="brute", weights=_weights)
     est.fit(X, y)
     assert_allclose(est.predict([[0, 2.5]]), [6])
+
+
+def test_nan_euclidean_support():
+    # Test input containing NaN.
```

Comment:
```suggestion
    """Check that the different neighbor estimators are lenient towards `nan`
    values if using `metric="nan_euclidean".
    """
```

Reply:
We should also make sure to check the output of predict. I would also add an additional test to check what happens if we have a full sample with `nan` values to see how it fails.

Reply:
Hello @glemaitre, thanks for the review! :D

Regarding your last comment, the all `nan` input leads to something similar to constant `X` (getting the first indexes as nearest neighbors) and being independent of `weights` (I was expecting something to happen when we weighed by distance).

```python
X = [[np.nan, np.nan], [np.nan, np.nan], [np.nan, np.nan], [np.nan, np.nan]]
y = [1, 2, 3, 4]

model = neighbors.KNeighborsClassifier(metric="nan_euclidean", n_neighbors=3, weights='uniform')
model.fit(X, y).predict(X)
>>> array([1, 1, 1, 1])

model = neighbors.KNeighborsClassifier(metric="nan_euclidean", n_neighbors=3, weights='distance')
model.fit(X, y).predict(X)
>>> array([1, 1, 1, 1])

X = [[0, 0], [0, 0], [0, 0], [0, 0]]
model = neighbors.KNeighborsClassifier(metric="nan_euclidean", n_neighbors=3, weights='distance')
model.fit(X, y).predict(X)
>>> array([1, 1, 1, 1])
```

Results are similar for other classifiers/estimators such as `RadiusNeighborsClassifier`.

For `KNeighborsTransformer` and `RadiusNeighborsTransformer` we get:
```python
model = neighbors.KNeighborsTransformer(metric="nan_euclidean", n_neighbors=2)
model.fit_transform(X).toarray()
>>> array([[0., 0., 0., 0.],
       [0., 0., 0., 0.],
       [0., 0., 0., 0.],
       [0., 0., 0., 0.]])
```

For `LocalOutlierFactor` we get:
```python
model = neighbors.LocalOutlierFactor(metric="nan_euclidean", n_neighbors=1)
model.fit_predict(X)
```

No errors are being released in any case. I think this makes sense for the estimators/neighbors search, but I'm unsure if we want this, especially in transformers. What do you think?

On the other hand, `pairwise_distances`' family of functions gives us nan distances:
```python
pairwise_distances(X, X, metric="nan_euclidean")
>>> array([[nan, nan, nan, nan],
       [nan, nan, nan, nan],
       [nan, nan, nan, nan],
       [nan, nan, nan, nan]])

pairwise_distances_argmin_min(X, X, metric="nan_euclidean")
>>> (array([0, 0, 0, 0], dtype=int64), array([nan, nan, nan, nan]))
```

Do you want me to do something about this?

Reply:
Unless possible new asserts/tests, like the one suggested here, I made the revisions you instructed, so I'm requesting a new review. :D Thanks for the first comments.